static/select_seats.js

```diff
@@ -0,0 +1,99 @@
+function setupSelectSeats(allSeats, takenSeats, pricings) {
+    var seats = document.querySelectorAll('.seat'),
+        booking = {};
+
+    function selectSeat(event) {
+        var seat = event.target.id;
+        if (booking.hasOwnProperty(seat)) {
+            removeSeat(seat);
+        } else {
+            addSeat(seat);
+        }
+    }
+
+    function addSeat(seat) {
+        var seatId = seat.replace("seat-", "");
+        booking[seat] = {id: seatId, value: null};
+        document.querySelector('#' + seat).classList.add('selected-seat');
+        renderBooking();
+    }
+
+    function removeSeat(seat) {
+        var seatId = seat.replace("seat-", "");
+        delete booking[seat];
+        document.querySelector('#' + seat).classList.remove('selected-seat');
+        renderBooking();
+    }
+
+    function selectSeatType(event) {
+        var target = event.target,
+            seatId = target.dataset.id,
+            seatType = target.value;
+        seatKey = "seat-" + seatId;
+
+        booking[seatKey].value = seatType;
+        renderBooking();
+    }
+
+    function renderBooking() {
+        output = "";
+        Object.keys(booking).forEach(function(seatId) {
+            output += renderSeatForm(booking[seatId]);
+        });
+
+        document.querySelector('#booking').innerHTML = output;
+        document.querySelectorAll('#booking select').forEach(function (select) {
```

Comment:
I though `querySelectorAll` returned a `NodeList` and not an `Array`. This should be `Array.prototype.forEach.call(document.querySelectorAll(...), (el) => {})`